### ISSUE TITLE
Autocompletion fundamentaly broken

### ISSUE THREAD

**Issue by MTRNord** [vector-im/element-web#18593]
### Steps to reproduce

Write a message with `:O` and see it getting autocompleted to the wrong emoji on enter.

### What happened?

Autocompletion gets `:v:` which is the peace sign and not the :O emoji.

### What did you expect?

The :O emoji

### Operating system

Windows

### Browser information

Firefox

### URL for webapp

develop.element.io

### Additional information

This is caused by https://github.com/matrix-org/matrix-react-sdk/pull/5659 which is a breaking change without warning changing autocompletion to the wrong key

**Comment by MTRNord** [comment]
To clarify my main issue is that `:O[ENTER]` does not as before send but now completes to a wrong emoji resulting in having to delete it again, writing `:O` again, adding a space and then I can send it. Which gets annoying to me. I would love to have a setting of tab vs enter. Imho Enter as autocomplete clashes with sending

**Comment by SimonBrandner** [comment]
> Imho Enter as autocomplete clashes with sending

And tab with ARIA standards...

Regarding the emoji issue: 

+ We should match emoticons no matter casing, I think - `:p` should be the same as `:P`
+ Perhaps when autocompleting emoji we should wait for the user to write two chars before we start showing suggestions to avoid stuff like this

**Comment by HarHarLinks** [comment]
>> Imho Enter as autocomplete clashes with sending

> And tab with ARIA standards...

the popup should tell you what keys to use, there is plenty space to put it:
![image](https://user-images.githubusercontent.com/2803622/129764556-70e474dd-0995-4615-a1b1-d82f0c959840.png)

"press [arrows] to select, [tabsymbol]  or [entersymbol]  to confirm, [escsymbol] to dismiss".

**Comment by SimonBrandner** [comment]
That is more of duplicate of https://github.com/vector-im/element-web/issues/17574

**Comment by robintown** [comment]
Upgrading this to S-Major as I've been realizing how annoying this is over the past couple days. You can't even send something as simple as ":)" without resorting to clicking the send button with your mouse :(

**Comment by SimonBrandner** [comment]
I've been thinking about this a bit and I've come to the following conclusion:

1. We should replace emoticons with emoji only after the message gets sent - this would be more consistent with Discord for example. An alternative would be to check if there is an emoticon at the end of the message and replace it if it is there. While that would be possible if feels a little inconsistent with itself and it could be a little surprising.
2. We shouldn't show emoticons in the emoji autocomplete - no other platform I know of does this and it is not very useful, especially with point 3.
3. We should display the emoji autocomplete only after the user types two letters (e.g. `:sm`) - this matches both Discord and Slack and feels saner. It avoids the current pain when you just want to type `:)` and a whole autocomplete appears. 

<hr>

I am not 100% sure how to do all this but I am happy to take a look if the design team gave their opinion on this

**Comment by ShadowJonathan** [comment]
+1 to point 3, I think that'd make the biggest difference, and also close a lot more issues

**Comment by novocaine** [comment]
Agree that fixing #3 is the right thing to do. It seems like that's an accidental regression according to feedback on https://github.com/vector-im/element-web/issues/17958

**Comment by dbkr** [comment]
There also seems to be a bug that's only hitting a small number of people where the `esc` key doesn't dismiss the autocomplete suggestions. Unfortunately none of those people are us, so we're at a complete loss on the cause.
